docs/docs/design/pvf-precheck.md

```diff
@@ -0,0 +1,154 @@
+```
+
+If the new PVF found subsystem initiate PVF pre-checking request adn wait for the result. Then based on the result it submits the vote.
+
+ThThe second runtime API is needed to submit the judgement for a PVF, whether it is approved or not. 
+The voting process uses unsigned transactions. The PvfCheckStatement is circulated through the network via gossip similar to a normal transaction. 
+At some point the validator will include the statement in the block, where it will be processed by the runtime. 
+If that was the last vote before gaining the super-majority, this PVF will not be returned by pvfs_require_precheck anymore.
+
+```rust
+fn submit_pvf_check_statement(stmt: PvfCheckStatement, signature: ValidatorSignature);
+```
+
+The vote will be distributed via the gossip similarly to a normal transaction. 
+Eventually a block producer will include the vote into the block where it will be handled by the runtime.
+
+
+One of the main units of information on which PVF pre-checking voting is build is the PvfCheckStatement.
+
+This is a statement by the validator who ran the pre-checking process for a PVF. A PVF is identified by the ValidationCodeHash.
+
+The statement is valid only during a single session, specified in the session_index.
+
+```rust
+struct PvfCheckStatement {
+    /// `true` if the subject passed pre-checking and `false` otherwise.
+    pub accept: bool,
+    /// The validation code hash that was checked.
+    pub subject: ValidationCodeHash,
+    /// The index of a session during which this statement is considered valid.
+    pub session_index: SessionIndex,
+    /// The index of the validator from which this statement originates.
+    pub validator_index: ValidatorIndex,
+}
+```
+
+
+There is no dedicated input mechanism for PVF pre-checker. Instead, PVF pre-checker looks on the ActiveLeavesUpdate event stream for work.
+The subsystem does not produce any output messages either, instead it sends messages to Runtime API  to query for the pending PVFs and to submit votes.
+In addition to that, it will also communicate with Candidate Validation Subsystem to request PVF pre-check.
+
+If the node is running in a collator mode, this subsystem will be disabled.
+
+The PVF pre-checker subsystem keeps track of the PVFs that are relevant for the subsystem. By  fetching them from runtime API 
+for every active leave.
+If the PVF is not present in any of the active leaves, it ceases to be relevant.
+PVF-precheck subsystem ask Candidate Validation subsystem to check on the PVF validity the result is submited to runtime API with
+`submit_pvf_check_statement`
+In case, a judgement was received for a PVF that is no longer in view it is ignored. << This probably means that if CV taking long ot response for PVF that is not any more in activel lives then we ignore it
+
+Whe session changes (at least one of the new active leave session index is > then prev on any of the leaves)  subsystem resign and submit only new session PVFs.
```

Comment:
What do you mean with "resign"? There is no mechanism to cancel to PVF pre-checking request sent to candidate validation, right?